cpp/basix/finite-element.h

```diff
@@ -837,6 +837,64 @@ class FiniteElement
       permute_data<std::int32_t, true>(d, 1, cell_info, _eperm_rev);
   }
 
+  /// @brief Permute indices associated with degree-of-freedoms on the
+  /// closure of a sub-entity of the reference element
+  ///
+  /// This function performs a similar permutation to permute() but
+  /// additionally permutes the positions of vertices and edges
+  ///
+  /// @note This function is designed to be called at runtime, so its
+  /// performance is critical.
+  ///
+  /// @param[in,out] d Indices associated with each reference element
+  /// degree-of-freedom (in). Indices associated with each physical
+  /// element degree-of-freedom (out).
+  /// @param cell_info Permutation info for the cell
+  /// @param entity_type The cell type of the sub-entity
+  void permute_subentity_closure(std::span<std::int32_t> d,
```

Comment:
It looks like `cell_info` is really the information for the sub-entity (e.g. if the cell was a tetrahedron and the sub-entity was an interval, you would have to pass data for the interval, which would be either 0 or 1 depending on if it is flipped or not). Do you think we could instead make `cell_info` the actual cell information (e.g. containing the information for all sub-entities of the tet, as is returned by `get_cell_permutation_info` in DOLFINx) and also pass a sub-entity index to identify which cell sub-entity. I think this might be easier than the user having to figure out the info for a particular sub-entity

Reply:
I've added `permute_subentity_closure_inv` that goes the other way and should be what you want and I've added an overloaded version of that takes `cell_into` and the entity index (as well as renaming the input in the original version to `entity_info`)